### PR TITLE
Add helm chart

### PR DESCRIPTION
**Pull request by MarcelMue**
towards https://github.com/giantswarm/giantswarm/issues/15992

### CODE REVIEW

File: helm/cluster-api-core/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      labels:
+        {{- include "labels.common" . | nindent 8 }}
+    spec:
+      containers:
+      - args:
+        - --secure-listen-address=0.0.0.0:8443
+        - --upstream=http://127.0.0.1:8080/
+        - --logtostderr=true
+        - --v=10
+        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.4.1
```
Comment (QuentinBisson) [inline helm/cluster-api-core/templates/deployment.yaml:26]: should we not retag here?

File: helm/cluster-api-core/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        imagePullPolicy: IfNotPresent
+        name: kube-rbac-proxy
+        ports:
+        - containerPort: 8443
+          name: https
+          protocol: TCP
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+      - args:
+        - --metrics-addr=127.0.0.1:8080
```
Comment (QuentinBisson) [inline helm/cluster-api-core/templates/deployment.yaml:36]: Should it be listening on localhost?

File: helm/cluster-api-core/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        - --feature-gates=MachinePool=false,ClusterResourceSet=false
```
[the remark this answers is not in the document]
Comment (MarcelMue) [reply helm/cluster-api-core/templates/deployment.yaml:37]: Mhm not sure. I will try this out some more to see.

File: helm/cluster-api-core/templates/deployment.yaml (hunk 1 above)
Comment (MarcelMue) [reply helm/cluster-api-core/templates/deployment.yaml:26]: I would say we retag as soon as we have some forks set up.

**Comment by MarcelMue** [comment]
@giantswarm/capi-hive-release-reliability Do you think anything is missing here or should I merge here for now?

**Review by anvddriesch — APPROVED** [review]
If we can deploy and run it, it should be fine for now.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  build:
+    jobs:
+      # Ensure that for every commit
+      # there is an app version in the test catalog.
+      - architect/push-to-app-catalog:
+          context: architect
+          name: push-to-app-catalog
+          app_catalog: "control-plane-test-catalog"
+          app_catalog_test: "control-plane-test-catalog"
+          chart: "cluster-api-core"
```
Comment (TheoBrigitte) [inline .circleci/config.yml:16]:
```
          filters:
            tags:
              only: /^v.*/
```

**Pull request merged**